env0/utils.go

```diff
@@ -71,13 +71,22 @@ func readResourceData(i interface{}, d *schema.ResourceData) error {
 			continue
 		}
 
+		// custom field is a pointer.
 		if customField, ok := field.Interface().(CustomResourceDataField); ok {
 			if err := customField.ReadResourceData(fieldNameSC, d); err != nil {
 				return err
 			}
 			continue
 		}
 
+		// custom field is a value, check the pointer.
```

Comment:
This is a small improvement to support custom serialization for non-pointer types.

Reply:
does it require additional tests? for both cases of pointer and non-pointer?

Reply:
pointer one is tested in both acceptance test and unit tests.
The non-pointer is tested in just acceptance test. I'll add another unit teat.

Reply:
Added a test see below.